File: TimeKeeperAndroid/src/com/tjoris/timekeeper/InputDialog.java
package com.tjoris.timekeeper;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

public abstract class InputDialog extends DialogFragment
{
	private final LayoutInflater fLayoutInflater;
	private final int fTitleID;
	private final int fIconID;
	private final int fViewID;
	private final int fPositiveTextID;
	private final int fNegativetextID;
	
	public InputDialog(final LayoutInflater layoutInflater, final int titleID, final int iconID, final int viewID, final int positiveTextID, final int negativeTextID)
	{
		fLayoutInflater = layoutInflater;
		fTitleID = titleID;
		fIconID = iconID;
		fViewID = viewID;
		fPositiveTextID = positiveTextID;
		fNegativetextID = negativeTextID;
	}
	
	@Override
	public Dialog onCreateDialog(final Bundle savedInstanceState)
	{
		final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
		final View view = fLayoutInflater.inflate(fViewID, null);
		viewCreated(view);
		builder.setTitle(fTitleID).setIcon(fIconID).setView(view).setPositiveButton(fPositiveTextID, new DialogInterface.OnClickListener()
		{
			public void onClick(final DialogInterface dialog, final int id)
			{
				dialogConfirmed(getDialog());
			}
		}).setNegativeButton(fNegativetextID, null);
		return builder.create();
	}
	
	public void viewCreated(final View view)
	{
	}
	
	public abstract void dialogConfirmed(Dialog dialog);
}

File: TimeKeeperAndroid/app/src/main/java/be/t_ars/timekeeper/InputDialog.kt
package be.t_ars.timekeeper

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.view.LayoutInflater
import android.view.View

class InputDialog : DialogFragment() {
    private var fViewCreatedAction: ((View) -> Unit)? = null
    private lateinit var fConfirmAction: ((View) -> Unit)
    private lateinit var fLayoutInflater: LayoutInflater
    private var fTitleID: Int = 0
    private var fIconID: Int = 0
    private var fViewID: Int = 0
    private var fPositiveTextID: Int = 0
    private var fNegativeTextID: Int = 0

    fun setOptions(viewCreatedAction: ((View) -> Unit)?, confirmAction: (View) -> Unit, layoutInflater: LayoutInflater, titleID: Int, iconID: Int, viewID: Int, positiveTextID: Int, negativeTextID: Int) {
        fViewCreatedAction = viewCreatedAction
        fConfirmAction = confirmAction
        fLayoutInflater = layoutInflater
        fTitleID = titleID
        fIconID = iconID
        fViewID = viewID
        fPositiveTextID = positiveTextID
        fNegativeTextID = negativeTextID
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
                .setTitle(fTitleID)
                .setIcon(fIconID)
        val view = fLayoutInflater.inflate(fViewID, null)
        fViewCreatedAction?.invoke(view)
        builder.setView(view)
                .setPositiveButton(fPositiveTextID) { _, _ -> fConfirmAction.invoke(view) }
                .setNegativeButton(fNegativeTextID, null)
        return builder.create()
    }
}
